Examples/Nodify.Shared/ThemeManager.cs

```diff
@@ -113,20 +163,31 @@ public static void SetTheme(string themeName)
             // Load new theme if it is valid
             if (_themesResources.TryGetValue(themeName, out var resources))
             {
-                foreach (var res in resources)
-                {
-                    Application.Current.Resources.MergedDictionaries.Add(res);
-                }
-
                 // Unload current theme
                 if (ActiveTheme != null)
                 {
                     foreach (var res in _themesResources[ActiveTheme])
                     {
                         Application.Current.Resources.MergedDictionaries.Remove(res);
+
+                        // Additionally search by Source:
+                        // as Theme resources in samples placed explicitly in App.xaml for xaml preview purposes
+                        // (however ThemeManager doesn't find those as already preloaded on startup,
+                        // because MainWindow not initialized yet)
+                        var duplicates = Application.Current.Resources.MergedDictionaries.Where(r => r.Source == res.Source).ToList();
```

Comment:
Can we detect duplicates in the `PreloadThemes` function? This may be the reason why changing themes is now slower than before (verified with ~400 nodes in Playground)

Reply:
its possible, but i guess the order is more important, not necessarily the duplication. The performance issue is likely related to Output Warning messages.